### ISSUE TITLE
Allow physical = false for players

### ISSUE THREAD

**Issue by appgurueu** [minetest/minetest#10139]
## Problem

The object property `physical = false` only works for entities and not for players.

## Solutions

`physical = false` should work for entities **and players**.

**Comment by sfan5** [comment]
So you want to be able to toggle noclip (and free_move?) from the server?

**Comment by paramat** [comment]
Why is this useful or needed?

**Comment by hecktest** [comment]
Players already don't collide with one another anyway.

**Comment by Zughy** [comment]
@appgurueu can you please address those questions?

**Comment by appgurueu** [comment]
> @appgurueu can you please address those questions?

You currently need to attach players to something to **force** them to be ghost-ish (at least in relation to physical `collide_with_objects` entities and blocks), which is **different** from the current noclip priv. This is obviously bad for controls (no client prediction). anon55555 put it quite well:

> A server might want to e.g. put a player into something like minecraft's spectator mode.

obviously this would be paired with e.g. turning off gravity through `set_physics`.

**Comment by Zughy** [comment]
how about an usecase though? Like, spectator mode doesn't work pretty well imho, because with `physical=false` you can't put invisible barriers to stop players from passing through (as they would be able to go through everything). You could check their position at every step, but it would also mean implementing map limits code-side.

**Comment by appgurueu** [comment]
> how about an usecase though? Like, spectator mode doesn't work pretty well imho, because with `physical=false` you can't put invisible barriers to stop players from passing through (as they would be able to go through everything). You could check their position at every step, but it would also mean implementing map limits code-side.

Fair point. If I remember correctly, my intended usecase originally was for corpses: I do not want entities (or other players) to collide with them. I ended up attaching the player to an entity for this purpose.

**Comment by Zughy** [comment]
can't you just do
```lua
player:set_properties({
collisionbox = {0}
}
```
when they die and restore it when they come back to life?

**Comment by appgurueu** [comment]
> can't you just do
> 
> ```lua
> player:set_properties({
>     collisionbox = {0}
> }
> ```

No. That's a "point" collisionbox (the one zero is odd BTW: either MT will fully default to zero or it won't default at all and you'd have to use 6 zeroes) and ought to not work the same as turning off collisions (see #10294). IIRC a "negative" sized collisionbox doesn't work either.
